### PR TITLE
feat: add Oathkeeper gRPC middleware

### PR DESCRIPTION
**Pull request by hperl**
This PR adds a gRPC middleware to Oathkeeper to use in other services.

### CODE REVIEW

**Review by zepatrik — CHANGES_REQUESTED** [review]
Nice, the API looks good :+1: 
Just some small improvements I'd suggest.

File: middleware/config.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package middleware
+
+import (
+	"github.com/ory/viper"
+	"github.com/ory/x/logrusx"
+)
+
+// initializeConfig initializes viper.
+func initializeConfig(l *logrusx.Logger, cfgFile string) {
+	viper.SetConfigFile(cfgFile)
```
[the remark this answers is not in the document]
Comment (hperl) [reply middleware/config.go:10]: Will be resolved by move to Koanf.

File: middleware/grpc_middleware.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	"net/url"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+var ErrDenied = status.Errorf(codes.Unauthenticated, "denied")
```
[the remark this answers is not in the document]
Comment (hperl) [reply middleware/grpc_middleware.go:16]: The problem is that this does not yet work for the version of herodot currently used in Oathkeeper. Will try again after the dependency bump.

Also, calling middlewares inside a middleware is not very clean, so I'd need to inline that functionality.

**Comment by hperl** [comment]
Closed in favor of https://github.com/ory/oathkeeper/pull/1004

**Pull request closed without merging**